### PR TITLE
[Support] Get RDS Instance quota from AWS API, and use that instead of hardcoded values.

### PR DESCRIPTION
**Pull request by whi-tw**
What
----

Pull the current quota from the AWS Service Quotas API, and use that as the limit against which the Quota alert is assessed.

How to review
-------------

Run down pipeline, ensure that the alert in dev stops being 'warning'

---

🚨⚠️ Please do not merge this pull request via the GitHub UI ⚠️🚨

### CODE REVIEW

**Review by AP-Hunt — CHANGES_REQUESTED** [review]
This should actually be changed in `default.yml` because in the Makefile, we have this line for the `dev` target

```$(eval export ENV_SPECIFIC_BOSH_VARS_FILE=default.yml)```

**Comment by whi-tw** [comment]
> This should actually be changed in `default.yml` because in the Makefile, we have this line for the `dev` target
> 
> `$(eval export ENV_SPECIFIC_BOSH_VARS_FILE=default.yml)`

Agreed technically, although we need to rethink this approach, as the default quota is 40, whereas we have had this increased up to 90.

**Comment by AP-Hunt** [comment]
Then let's consider doing the work to correct this.

**Comment by LeePorte** [comment]
Sounds like a sensible thing to fix to me

**Comment by whi-tw** [comment]
The alternative is to use the Quota API to pull in the current limit as a data point and use that for the alert. I've got some of the work for that done already in a branch, so I'll have a look - that'd be a much more reliable way of doing this, rather than the manual syncing we're doing right now.

**Review by AP-Hunt — CHANGES_REQUESTED** [review]
The pattern in `paas-metrics` for working with AWS services is to wrap them up in our own helper libraries, and test those. It would be good to do the same here for consistency.

**Comment by whi-tw** [comment]
> The pattern in `paas-metrics` for working with AWS services is to wrap them up in our own helper libraries, and test those. It would be good to do the same here for consistency.

Confused. That's what I've done, right? I followed the same pattern as with the rds API

File: tools/metrics/gauges_rds_db_instances.go [hunk 1]
```diff
@@ -50,3 +66,22 @@ func countRDSDBInstance(service *rds.RDSService) (int, error) {
 
 	return total, nil
 }
+
+func getRDSServiceQuota(service *servicequotas.ServiceQuotas) (float64, error) {
+	out, err := service.Client.GetServiceQuota(&awsservicequotas.GetServiceQuotaInput{
+		QuotaCode:   aws.String("L-7B6409FD"),
```
[the remark this answers is not in the document]
Comment (whi-tw) [reply tools/metrics/gauges_rds_db_instances.go:72]: The problem is, there's no real good reference to that string - for some reason, there are no results even if you google it. However, if you go to the URL [I've linked in the alert](https://console.aws.amazon.com/servicequotas/home?region=((aws_region))#!/services/rds/quotas/L-7B6409FD) it's definitely the right one. How would you suggest doing this?

[the remark this answers is not in the document]
Comment (AP-Hunt) [reply tools/metrics/gauges_rds_db_instances.go:72]: I think extract the constant and add just that explanation alongside it

**Comment by AP-Hunt** [comment]
Err, yes it is. Sorry! I got confused between `awsservicequotas` and `servicequotas`. I was also expecting to see some changes to the `pkg/` directory in the main commit, having not realised you'd added that bit to the vendoring commit. My bad.

**Review by AP-Hunt — APPROVED** [review]

**Comment by whi-tw** [comment]
> Err, yes it is. Sorry! I got confused between `awsservicequotas` and `servicequotas`. I was also expecting to see some changes to the `pkg/` directory in the main commit, having not realised you'd added that bit to the vendoring commit. My bad.

No trouble! Easily done! I wanted both commits to work independently and make sense, so I split them up in this way! I'll get this merged in!

**Pull request merged**